server/lib/device/light/light.command.js

```diff
@@ -55,6 +62,9 @@ async function command(message, classification, context) {
     logger.debug(e);
     this.messageManager.replyByIntent(message, 'light.turn-on.fail', context);
   }
+  if (nbLightsAffected === 0) {
```

Comment:
There is a problem with this approach, here it sends a "light.not-found" message when the turn on fails, because the response is outside of the try ! 

<img width="893" alt="Screenshot 2020-11-20 at 16 05 39" src="https://user-images.githubusercontent.com/7365207/99809024-6828a300-2b4a-11eb-97a9-01ff323f8b8e.png">

It should be the last step of the try :)

Reply:
My bad, fixed.
What is your prefered approach for testing ? Mocking or create data for use cases to test ?

Reply:
Thanks! 

It depends of what you are testing. Mocking is useful to test isolated code without depending of the rest of the stack. Creating data is more useful for E2E tests. Both are great!